UM/CentralFileStorage.py

```diff
@@ -148,6 +149,7 @@ def _hashFile(cls, file_path: str) -> str:
             while len(contents) > 0:
                 hasher.update(contents)
                 contents = f.read(block_size)
+        QCoreApplication.processEvents()  # Process events to allow the interface to update
```

Comment:
Alternatively; you could also call the self._application.showSplashMessage, since that ensures that processEvents is being called.